recordm/customUI/dash/src/components/Calendar.vue

```diff
@@ -309,10 +375,99 @@
           this.calendarApi.changeView(newContent.activeView ? newContent.activeView : this.calendarOptions.initialView )
         }
       },
+      getCustomButtomObject(viewThisObject){
+        if (viewThisObject.calendarOptions["customButtons"]){
+          return viewThisObject.calendarOptions["customButtons"]
+        }
+        function dateChanges(prev,isToday) {
+          const previousDateElement = viewThisObject.datePickerElement.children[0]
+          const nextDateElement = viewThisObject.datePickerElement.children[2]
+
+          let prevDate, nextDate
+
+          prevDate = new Date(previousDateElement.value)
+          nextDate = new Date(nextDateElement.value)
+          
+          let diff = nextDate.getTime() - prevDate.getTime()
 
+          if(diff == 0){
+            diff = nextDate.getTime() - (new Date(nextDate.getFullYear(),nextDate.getMonth(),nextDate.getDate()-1))
+          }
+
+          diff = prev ? diff*(-1):diff
+
+          prevDate = new Date(prevDate.getTime()+diff)
+          nextDate = new Date(nextDate.getTime()+diff)
+
+          if(isToday){
+            prevDate = new Date()
+            nextDate = new Date(prevDate.getFullYear()+1,nextDate.getMonth(),nextDate.getDate())
+          } 
+          previousDateElement.value = getInputTypeOfDate(prevDate)
+          nextDateElement.value = getInputTypeOfDate(nextDate)
+
+          viewThisObject.dateRange = [prevDate, nextDate]
+        }
```

Comment:
Tudo isto precisa de comentários. Preciso que me expliques os casos que estás a considerar.

Reply:
O nome do método está mal. Essa função reage ao evento de click dos botões PREV, NEXT, e TODAY.

Reply:
> O nome do método está mal. Essa função reage ao evento de click dos botões PREV, NEXT, e TODAY.

Isso eu entendi. O que não entendi foi a lógica, ou seja, os casos que estás a considerar.

Reply:
diff = prev ? diff*(-1):diff -> se for o botao prev subtrai-se a diferença, se for next soma-se
se for today, 
if(isToday){
            prevDate = new Date()
            nextDate = new Date(prevDate.getFullYear()+1,nextDate.getMonth(),nextDate.getDate())
          } .... Vou refazer o codigo